python/calico/felix/profilerules.py

```diff
@@ -246,9 +258,11 @@ def _delete_chains(self):
         """
         # Need to block here: have to wait for chains to be deleted
         # before we can decref our ipsets.
-        self._iptables_updater.delete_chains(
-            self.iptables_generator.profile_chain_names(self.id),
-            async=False)
+        chain_names = self.iptables_generator.profile_chain_names(self.id)
+        if self._untracked:
```

Comment:
Yes, for example, at https://github.com/neiljerram/felix/blob/untracked/python/calico/felix/endpoint.py#L901...

Or have I misunderstood your idea here?